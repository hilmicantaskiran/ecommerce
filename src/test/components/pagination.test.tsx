import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Pagination from '../../components/paginaton';

describe('Pagination', () => {
    it('should render correctly', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <Pagination
                    pages={100}
                    setCurrentPage={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    }),
    it('should render correctly with no pages', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <Pagination
                    pages={0}
                    setCurrentPage={() => {}}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
