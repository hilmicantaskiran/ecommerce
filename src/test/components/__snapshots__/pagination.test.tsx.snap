// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
<DocumentFragment>
  <div
    class="my-10"
  >
    <ul
      class="inline-flex -space-x-px"
    >
      <li
        class="disabled"
      >
        <a
          class="block py-2 px-3 ml-0 leading-tight text-gray-500"
          href="#!"
        >
          <span
            class="sr-only"
          >
            Previous
          </span>
          <svg
            aria-hidden="true"
            class="w-4 h-4"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </li>
      <li
        class="mt-1 bg-white rounded-lg"
      >
        <a
          class="py-2 px-3 leading-tight text-[#2a59fe]"
          href="#!"
        >
          1
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          2
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          3
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          4
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          5
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          6
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          7
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          8
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          9
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          10
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          11
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          12
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          13
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          14
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          15
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          16
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          17
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          18
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          19
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          20
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          21
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          22
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          23
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          24
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          25
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          26
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          27
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          28
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          29
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          30
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          31
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          32
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          33
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          34
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          35
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          36
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          37
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          38
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          39
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          40
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          41
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          42
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          43
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          44
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          45
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          46
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          47
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          48
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          49
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          50
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          51
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          52
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          53
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          54
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          55
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          56
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          57
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          58
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          59
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          60
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          61
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          62
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          63
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          64
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          65
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          66
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          67
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          68
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          69
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          70
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          71
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          72
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          73
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          74
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          75
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          76
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          77
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          78
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          79
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          80
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          81
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          82
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          83
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          84
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          85
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          86
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          87
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          88
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          89
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          90
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          91
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          92
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          93
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          94
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          95
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          96
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          97
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          98
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          99
        </a>
      </li>
      <li
        class="mt-1"
      >
        <a
          class="py-2 px-3 leading-tight text-gray-500 hover:text-black"
          href="#!"
        >
          100
        </a>
      </li>
      <li
        class=""
      >
        <a
          class="block py-2 px-3 ml-0 leading-tight text-gray-500"
          href="#!"
        >
          <span
            class="sr-only"
          >
            Next
          </span>
          <svg
            aria-hidden="true"
            class="w-4 h-4"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Pagination should render correctly with no pages 1`] = `
<DocumentFragment>
  <div
    class="my-10"
  >
    <ul
      class="inline-flex -space-x-px"
    >
      <li
        class="disabled"
      >
        <a
          class="block py-2 px-3 ml-0 leading-tight text-gray-500"
          href="#!"
        >
          <span
            class="sr-only"
          >
            Previous
          </span>
          <svg
            aria-hidden="true"
            class="w-4 h-4"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </li>
      <li
        class=""
      >
        <a
          class="block py-2 px-3 ml-0 leading-tight text-gray-500"
          href="#!"
        >
          <span
            class="sr-only"
          >
            Next
          </span>
          <svg
            aria-hidden="true"
            class="w-4 h-4"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
